src/context/ChainApiContext.tsx

```diff
@@ -84,7 +90,7 @@ export function ChainApiContextProvider({children}: {children: React.ReactNode})
       apiPromise.off('error', handleError);
       clearInterval(retryInterval);
     };
-  }, [currentNetwork.ws, state.wsConnectionIndex]);
+  }, [currentNetwork.ss58Format, currentNetwork.ws, state.wsConnectionIndex]);
```

Comment:
NIT: We might as well just put the `currentNetwork` object here.
```suggestion
  }, [currentNetwork, state.wsConnectionIndex]);
```